user.py

```diff
@@ -32,33 +48,84 @@ def add_user():
     name = request.form['name']
     first_name = request.form['first_name']
     email = request.form['email']
-
     if not contains_valid_characters(name):
         return make_response("Invalid characters in name", 400)
     if not contains_valid_characters(first_name):
         return make_response("Invalid characters in first name", 400)
     if email != '' and not is_valid_email(email):
         return make_response("Invalid email format", 400)
 
+    organization_code = request.form['organization_code']
+    is_teacher = 'is_teacher' in request.form
+    is_researcher = 'is_researcher' in request.form
+    supervisor_id = request.form.get('supervisor') if is_researcher else None
+    researcher_type = request.form['researcher_type'] if is_researcher else None
+
     try:
-        new_user = User(name=name, first_name=first_name, email=email)
+        new_user = User(name=name, first_name=first_name, email=email, is_teacher=is_teacher,
+                        is_researcher=is_researcher, supervisor_id=supervisor_id,
+                        organization_id=organization_code)
         db.session.add(new_user)
         db.session.commit()
+        if is_researcher:
```

Comment:
I suspect you need to keep such a field in the `User` model to do this. Actually, creating a teacher or researcher profile can be done in two steps if needed, or just be processed at runtime and not stored in database.

Reply:
I spoke to Alexandre about how to implement this problem and he advised me that when I create a user and check its role, I create a tuple in the db in this way: 

If it's a teacher: I create a tuple that only contains the user id linked to the time that the user is assigned to a lesson. 
If it's a researcher: I create the researcher directly because I don't need any specific conditions

In addition to this, I've added an 'is_active' field in the researcher and teacher table, which means I don't have to delete the row in the db every time I check/uncheck the teacher/researcher box.

Does this solution seem right to you or is there a simpler way of solving this problem?

Reply:
Samuel, I don't think I understand your explanation. What do you mean by tuple ? 

Essentially, a teacher/researcher is created in DB for every course the user is a teacher/researcher for, right ? So you don't need to create a Teacher or Researcher instance in the DB right away. That can be done when the user is assigned to a course (as teacher or researcher). 
Storing "is_researcher" and "is_teacher" in the DB allows us to give access to certain features/pages in the app. And that can later be changed to roles. Hadn't we discussed those flags in a previous meeting ?

Reply:
I misunderstood Anthony's message and got everything mixed up in my code, so you can ignore my message.

Reply:
For me there is still no need to keep `is_teacher` or `is_researcher` fields in the `User` model. You know a user is a teacher or researcher (or...both) if a `Teacher` or `Researcher` exists in database. SQLAlchemy relationships should handle this. Otherwise you need to keep the cosnsitency manually and check that the field in `User` table is correct when a profile is deleted.

This is not blocking. We can go forward with this right now.